### PR TITLE
Define trait Persist: interface for saving/restoring a component from state

### PR DESCRIPTION
**Pull request by sandreim**
## Reason for This PR

Implements #1715

## Description of Changes

Provided in commit message.

- [x] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any API changes are reflected in `firecracker/swagger.yaml`.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.

### CODE REVIEW

File: src/snapshot/src/persist.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//! Defines an abstract interface for saving/restoring a component from state.
+#![deny(missing_docs)]
```
Comment (andreeaflorescu) [inline src/snapshot/src/persist.rs:4]: We should either add this to the `lib.rs` level, or remove the exclamation mark. The exclamation mark says that this macro applies globally, at the crate level. You typically add this in `src/lib.rs`. If we want to only deny the missing docs in this file, the exclamation mark is not needed.

File: src/snapshot/src/persist.rs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+/// An abstract interface for saving/restoring a component from state.
```
Comment (andreeaflorescu) [inline src/snapshot/src/persist.rs:6]: nit: we can replace "from state" with "using its defined state". "from" only makes sense for restoring, and not for saving.

File: src/snapshot/src/persist.rs [hunk 3]
```diff
@@ -0,0 +1,25 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
```
Comment (lauralt) [inline src/snapshot/src/persist.rs:2]: nit: one more newline.

File: src/snapshot/src/persist.rs (hunk 2 above)
Comment (sandreim) [reply src/snapshot/src/persist.rs:6]: Yep, I'll update.

File: src/snapshot/src/persist.rs (hunk 1 above)
Comment (sandreim) [reply src/snapshot/src/persist.rs:4]: It is present in lib.rs. I'll remove it from here.

**Pull request merged**